integration/simple-nullprototypes/simple-test.ts

```diff
@@ -0,0 +1,249 @@
+import { Reader } from 'protobufjs';
+import { Child_Type, Nested, Nested_InnerEnum, OneOfMessage, Simple, SimpleWithMap, StateEnum } from './simple';
+import { simple as pbjs, google } from './pbjs';
+import ISimple = pbjs.ISimple;
+import PbChild = pbjs.Child;
+import PbSimple = pbjs.Simple;
+import PbSimpleWithMap = pbjs.SimpleWithMap;
+import PbState = pbjs.StateEnum;
+import PbNested = pbjs.Nested;
+import PbNested_InnerMessage = pbjs.Nested.InnerMessage;
+import PbNested_DeepMessage = pbjs.Nested.InnerMessage.DeepMessage;
+import PbNested_InnerEnum = pbjs.Nested.InnerEnum;
+import INested = pbjs.INested;
+import PbTimestamp = google.protobuf.Timestamp;
+import * as Long from 'long';
+
+const jan1 = new Date('1970-01-01T00:00:00.000Z');
+
+describe('simple', () => {
+  it('generates types correctly', () => {
+    const simple: Simple = {
+      name: 'asdf',
+      age: 1,
+      child: { name: 'child', type: Child_Type.UNKNOWN },
+      state: StateEnum.ON,
+      grandChildren: [
+        { name: 'grand1', type: Child_Type.UNKNOWN },
+        { name: 'grand2', type: Child_Type.UNKNOWN }
+      ],
+      coins: [2, 4, 6],
+      snacks: ['a', 'b'],
+      oldStates: [StateEnum.ON, StateEnum.OFF],
+      createdAt: jan1,
+      thing: undefined,
+      blobs: []
+    };
+    expect(simple.name).toEqual('asdf');
+  });
+
+  it.only('can decode', () => {
+    const s1: ISimple = {
+      name: 'asdf',
+      age: 1,
+      createdAt: undefined,
```

Comment:
I found some strange behaviour here that I don't understand enough about protobuf to explain.

In the original simple test createdAt, thing and the child.type values are all missing.
I am not sure if this is acceptable or a slightly variation in functionality that just needs to be documented in the options

Reply:
Looks like some of these are fixed by https://github.com/stephenh/ts-proto/commit/7b839834dfa176850132c87322623f5348487704